pkg/api/v1/schema.go

```diff
@@ -32,6 +32,22 @@ const (
 	IOThreadsPolicyAuto   IOThreadsPolicy = "auto"
 )
 
+// Represents the firmware blob used to assist in the domain creation process.
+// Used by Xen fully virtualized domains as well as setting the QEMU BIOS file path for QEMU/KVM domains
+// ---
+// +k8s:openapi-gen=true
+type Bootloader struct {
+	// The path to the firmware blob
+	Path string `json:"path"`
```

Comment:
For our public API, I want to make sure we are presenting the simplest most concise options to the user. I don't think we need all of these options. For example, "type" will always be pflash for uefi, so why even expose that option? And "path" is something we know as well, so why explicitly expose that option? ReadyOnly vs writable is a feature that this PR doesn't implement, so we could drop that option and only support ReadOnly for now.

Instead of replicating the libvirt domain xml in our public KubeVirt api, can we consider just a 'bootloader' enum now that toggles between bios/uefi? Based on what the user chooses we can fill in all the details about the domain xml during the VMI->LibvirtDomainXml conversion.

Reply:
I sort of figured after [the initial reviews](https://github.com/kubevirt/kubevirt/pull/1814#discussion_r239418197) pointing out that Secure boot functionality might be wanted it would be better to expose most of these options. I'm happy to change this API to however you guys would like it represented; I don't really have any strong feelings on it, I just need the functionality. :)

Definitely agree on having the simplest options presented to the user though. But, as I mentioned earlier I'm a bit out of my domain here so I don't really know what the most common/best/correct options are that should be exposed.

Reply:
@michalskrivanek I'm guessing you likely have the best understanding of how the bootloader functionality is used in practice. What are your thoughts on the most concise way to represent this feature from the limited KubeVirt api perspective?

Reply:
the bootloader needs to be part of the libvirt/qemu container so it doesn't necessarily make sense to have it fully configurable (if anyone would want to extend kubevirt and ship theirs I suppose hooks would be used). So I think limiting(simplifying) the API to only those 2 firmwares OVMF ships with would be best.

Reply:
the nvram is more tricky shipped by the OVMF package for initial boot, and only then ideally persist it when guest writes to it. Practically it works good enough without persistence and we can also jsut always start with a blank template for now

Reply:
> the bootloader needs to be part of the libvirt/qemu container so it doesn't necessarily make sense to have it fully configurable (if anyone would want to extend kubevirt and ship theirs I suppose hooks would be used). So I think limiting(simplifying) the API to only those 2 firmwares OVMF ships with would be best.
@michalskrivanek how many bootloaders can we expect? is it usually really just one and we can just place it in the container, or various versions can be expected as a real use case, and then we'd want to place it in its own container or PV to be used with the libvirt/qemu container?

Reply:
I think just those two ("normal" and secure boot) and corresponding nvram (without or with keys). I can imagine eventually someone would want to roll out their own key and then we'd have to support custom nvram or ability to persist it....but it sounds pretty uncommon right now. More details in https://bugzilla.redhat.com/show_bug.cgi?id=1561128

Reply:
> I can imagine eventually someone would want to roll out their own key and then we'd have to support custom nvram or ability to persist it.

If that ends up happening, we have a hooks available to users that let them achieve this on a technical level. So we are not completely boxing ourselves in here.

Reply:
Hi @davidvossel, could you review these latest changes and see if they meet your requirements? Thanks!